staging/src/k8s.io/controller-manager/pkg/apis/config/v1alpha1/register.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// GroupName is the group name use in this package
+const GroupName = "controllermanager.config.k8s.io"
```

Comment:
this is just a bucket for shared types. We don't need this file at all.

Reply:
I'm not sure here, little confusion.
From KEP, `GenericControllerManagerConfiguration` is nor real API, but in `GenericControllerManagerConfiguration` struct, add `metav1.TypeMeta`, which confued me

Reply:
Having `TypeMeta` in `GenericControllerManagerConfiguration` sounds wrong, I agree.